unity/unity-embed-host/CoreCLRHost.cs

```diff
@@ -95,6 +95,34 @@ public static IntPtr gchandle_get_target_v2([NativeWrapperType("uintptr_t")] Int
         return Unsafe.As<object, IntPtr>(ref obj);
     }
 
+    [return: NativeCallbackType("MonoObject*")]
+    public static IntPtr object_isinst([NativeCallbackType("MonoObject*")] IntPtr obj, [NativeCallbackType("MonoClass*")] IntPtr klass)
+    {
+        var instance = Unsafe.As<IntPtr, object>(ref obj);
+        var type = Unsafe.As<IntPtr, Type>(ref klass);
```

Comment:
`klass` will be a pointer to native metadata not a `System.Type` instance. It should be compatible with RuntimeTypeHandle if things are setup correctly.

So you want something like:
```
Type.GetTypeFromHandle(RuntimeTypeHandle.FromIntPtr(klass))
```